### PR TITLE
feat: Add basic Group / cohort Notebok nodes

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

Nothing too fancy, but wanted to at least have some sort of representation for groups / cohorts in a Notebook.

## Changes

* Some light refactoring of the group logic
* Cohort and Group node, at least with actions to see more data
* No editing or anything advanced yet.

Screenshot 2023-10-13 at 10 56 51

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    return (
+        <div className="flex flex-col overflow-hidden">
+            <div className={clsx('p-4 flex-0 flex gap-2 justify-between')}>
+                {groupDataLoading ? (
+                    <LemonSkeleton className="h-6" />
+                ) : groupData ? (
+                    <>
+                        <GroupCaption groupData={groupData} groupTypeName={groupTypeName} />
+                    </>
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx:76]:
```suggestion
                    <GroupCaption groupData={groupData} groupTypeName={groupTypeName} />
```

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeCohort.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    const title = cohort ? `Cohort: ${cohort.name}` : 'Cohort'
+
+    const modifiedQuery = useMemo<DataTableNode>(() => {
+        return {
+            ...query,
+            embedded: true,
+            // TODO: Add back in controls in a way that actually works...
+            full: false,
+            showElapsedTime: false,
+            showTimings: false,
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeCohort.tsx:35]: We should hide the editor button here too

```suggestion
            showTimings: false,
            showOpenEditorButton: false,
```

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeCohort.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+    const { expanded } = useValues(notebookNodeLogic)
+    const { setExpanded, setActions, insertAfter } = useActions(notebookNodeLogic)
+
+    const { cohort, cohortLoading, cohortMissing, query } = useValues(cohortEditLogic({ id }))
+    const { setQuery } = useActions(cohortEditLogic({ id }))
+
+    const title = cohort ? `Cohort: ${cohort.name}` : 'Cohort'
+
+    const modifiedQuery = useMemo<DataTableNode>(() => {
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeCohort.tsx:28]: Worry a little bit about divergence here with `NotebookNodeQuery`. Probably something we can look at in the future but we should consider sharing the properties set on the `query` given they will probably match

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                                        },
+                                    ],
+                                },
+                            },
+                        },
+                    })
+                },
+            },
+        ])
+    }, [groupData])
```
Comment (daibhin) [inline frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx:62]: I'm assuming this value changes whenever the `title` or `groupTypeIndex` does too

**Review by daibhin — APPROVED** [review]
Looks good, nothing blocking but a couple of things to consider:

1) Should we include the "Add to Notebook" button on individual pages for cohorts and groups?

2) Do we need to add these to the slash commands? Maybe not until editing is possible inside a notebook

File: frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx (hunk 4 above)
Comment (benjackwhite) [reply frontend/src/scenes/notebooks/Nodes/NotebookNodeGroup.tsx:62]: This will be improved in the editable titles PR so will leave for now

**Comment by benjackwhite** [comment]
> 1. Should we include the "Add to Notebook" button on individual pages for cohorts and groups?

Yep good point

> 2. Do we need to add these to the slash commands? Maybe not until editing is possible inside a notebook

I don't think so. Currently the slash commands are more like "queryable things", whereas these are like indivudal nodes (more what should be inserted from global search)

**Pull request merged**